pipelines/common/src/test/java/org/openmrs/analytics/LocalDwhFilesTest.java

```diff
@@ -95,6 +106,50 @@ public void copyResourceTypeTest() throws IOException {
     Files.delete(sourcePath);
   }
 
+  @Test
+  public void writeTimestampFile_FileAlreadyExists_ThrowsError() throws IOException {
+    Path root = Files.createTempDirectory("DWH_FILES_TEST");
+    Path timestampPath = Paths.get(root.toString(), "timestamp.txt");
+    createFile(timestampPath, Instant.now().toString().getBytes(StandardCharsets.UTF_8));
+    DwhFiles dwhFiles = new DwhFiles(root.toString(), FhirContext.forR4Cached());
+
+    Assert.assertThrows(FileAlreadyExistsException.class, () -> dwhFiles.writeTimestampFile());
+
+    Files.delete(timestampPath);
+    Files.delete(root);
+  }
+
+  @Test
+  public void writeTimestampFile_FileDoesNotExist_CreatesFile() throws IOException {
+    Path root = Files.createTempDirectory("DWH_FILES_TEST");
+    DwhFiles dwhFiles = new DwhFiles(root.toString(), FhirContext.forR4Cached());
+
+    dwhFiles.writeTimestampFile();
+
+    List<Path> destFiles = Files.list(root).collect(Collectors.toList());
+    assertThat(destFiles.size(), equalTo(1));
+    assertThat(destFiles.get(0).toString(), equalTo(root.resolve("timestamp.txt").toString()));
+
+    Files.delete(destFiles.get(0));
+    Files.delete(root);
+  }
+
+  @Test
+  public void readTimestampFile() throws IOException {
+    Path root = Files.createTempDirectory("DWH_FILES_TEST");
+    Instant currentInstant = Instant.now();
+    Path timestampPath = Paths.get(root.toString(), "timestamp.txt");
+    createFile(timestampPath, currentInstant.toString().getBytes(StandardCharsets.UTF_8));
+    DwhFiles dwhFiles = new DwhFiles(root.toString(), FhirContext.forR4Cached());
+
+    Instant actualInstant = dwhFiles.readTimestampFile();
+
+    Assert.assertEquals(currentInstant.getEpochSecond(), actualInstant.getEpochSecond());
```

Comment:
nit: I would prefer Hamcrest's assertions over JUnit; one part is consistency with everywhere else in our tests but the main reason is that in general Hamcrest gives better error messages when the assertion fails AFAIK.

Reply:
Noted